tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,184 @@
+
+    add_allow_all_policy_to_yaml "${kata_pod_with_signed_image}"
+}
+
+@test "Create a pod from an unsigned image, on an insecureAcceptAnything registry works" {
+    # We want to set the default policy to be reject to rule out false positives
+    setup_kbs_image_policy "reject"
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_UNPROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' is rejected" {
+    # We want to leave the default policy to be insecureAcceptAnything to rule out false negatives
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: Cannot pull manifest"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry' is successful" {
+    # We want to set the default policy to be reject to rule out false positives
+    setup_kbs_image_policy "reject"
+
+    create_pod_yaml_with_signed_image "${COSIGN_SIGNED_PROTECTED_REGISTRY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+@test "Create a pod from a signed image, on a 'restricted registry', but with the wrong key is rejected" {
+    # We want to leave the default policy to be insecureAcceptAnything to rule out false negatives
+    setup_kbs_image_policy
+
+    create_pod_yaml_with_signed_image "${COSIGNED_SIGNED_PROTECTED_REGISTRY_WRONG_KEY_IMAGE}"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    assert_pod_fail "${kata_pod_with_signed_image}"
+    assert_logs_contain "${node}" kata "${node_start_time}" "Security validate failed: Validate image failed: \[PublicKeyVerifier"
+}
+
+@test "Create a pod from an unsigned image, on a 'restricted registry' works if enable_signature_verfication is false" {
+    # We want to set the default policy to be reject to rule out false positives
+    setup_kbs_image_policy "reject"
+
+    create_pod_yaml_with_signed_image "${UNSIGNED_PROTECTED_REGISTRY_IMAGE}" "false"
+
+    # For debug sake
+    echo "Pod ${kata_pod_with_signed_image}: $(cat ${kata_pod_with_signed_image})"
+
+    k8s_create_pod "${kata_pod_with_signed_image}"
+    echo "Kata pod test-e2e from image security policy is running"
+}
+
+teardown() {
+    if [ "${KATA_HYPERVISOR}" = "qemu-tdx" ]; then
```

Comment:
Hi @stevenhorsman !

[Latest execution on TDX](https://github.com/kata-containers/kata-containers/actions/runs/10305807718/job/28563027264?pr=10009) had signature tests passing, except for the first one:  `not ok 1 Create a pod from an unsigned image, on an insecureAcceptAnything registry works`. But the exit status of k8s-guest-pull-image-signature.bats became zero because of this skip. If we want tests skipped on TDX then it should be skipped on `setup()` too. is it the case?

If you end up updating this commit, can you fix the "verificaiton" typo on the commit msg?

Reply:
Sorry, I missed removing that. I'll tidy it up now and then can always add TDX skip as a separate commit if needed